### PR TITLE
Clientizen events system

### PR DESCRIPTION
**Pull request by tal5**

### CODE REVIEW

File: src/main/java/com/denizenscript/depenizen/bukkit/clientizen/Channels.java [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 public class Channels {
     public static final String CHANNEL_NAMESPACE = "clientizen";
-    public static final String SET_SCRIPTS = CHANNEL_NAMESPACE + ":" + "set_scripts";
-    public static final String RECEIVE_CONFIRM = CHANNEL_NAMESPACE + ":" + "receive_confirmation";
+    public static final String SET_SCRIPTS = CHANNEL_NAMESPACE + ':' + "set_scripts";
+    public static final String RECEIVE_CONFIRM = CHANNEL_NAMESPACE + ':' + "receive_confirmation";
+    public static final String EVENT_DATA = CHANNEL_NAMESPACE + ':' + "event_data";
+    public static final String RECEIVE_EVENT = CHANNEL_NAMESPACE + ':' + "fire_event";
 }
```
Comment (acikek) [inline src/main/java/com/denizenscript/depenizen/bukkit/clientizen/Channels.java]: Not sure what's going on with these strings, feel like creating them should be a method like in Clientizen

File: src/main/java/com/denizenscript/depenizen/bukkit/clientizen/events/ClientizenScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public void fireInternal(Player player, DataDeserializer data) {
+        this.player = player;
+        fire(data);
+    }
+
+    public void fire(DataDeserializer data) {}
+
+    public void write(DataSerializer serializer) {}
```
Comment (acikek) [inline src/main/java/com/denizenscript/depenizen/bukkit/clientizen/events/ClientizenScriptEvent.java]: Same abstract comment for these

File: src/main/java/com/denizenscript/depenizen/bukkit/clientizen/Channels.java (hunk 1 above)
Comment (tal5) [reply src/main/java/com/denizenscript/depenizen/bukkit/clientizen/Channels.java]: True, added a `Channels#id` method

File: src/main/java/com/denizenscript/depenizen/bukkit/clientizen/events/ClientizenScriptEvent.java (hunk 2 above)
Comment (tal5) [reply src/main/java/com/denizenscript/depenizen/bukkit/clientizen/events/ClientizenScriptEvent.java]: Added a comment to clarify

**Pull request merged**